pkg/cmd/step/helm/step_helm_apply.go

```diff
@@ -503,7 +503,7 @@ func defineAppsChartOverridingError(chartName string, err error) error {
 func (o *StepHelmApplyOptions) fetchSecretFilesFromVault(dir string, store configio.ConfigStore) ([]string, error) {
 	log.Logger().Debugf("Fetching secrets from vault into directory %q", dir)
 	files := []string{}
-	client, err := o.SystemVaultClient(kube.DefaultNamespace)
+	client, err := o.SystemVaultClient("")
```

Comment:
This will look up dynamically the dev namespace, which will allow a bit more flexibility. It seems like a good change.